src/librustc_privacy/lib.rs

```diff
@@ -974,7 +974,7 @@ impl<'a, 'tcx: 'a> TypeVisitor<'tcx> for SearchInterfaceForPrivateItemsVisitor<'
         if let ty::TyProjection(ref proj) = ty.sty {
             // Avoid calling `visit_trait_ref` below on the trait,
             // as we have already checked the trait itself above.
-            proj.trait_ref.super_visit_with(self)
+            proj.substs.visit_with(self)
```

Comment:
I *think* I got it, see if that makes sense (may not even compile, but at least it's not immediately failing).